### ISSUE TITLE
`--check-str-concat-over-line-jumps=no` doesn't work for calls, just assignments

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#6515]
### Bug description

According to https://github.com/PyCQA/pylint/issues/6441#issuecomment-1107449955, this was supposed to work before we extended this to calls:

```python
import unittest

class MyTest(unittest.TestCase):
    def testColors(self):
        self.assertEqual(
            'red',
            'A.color = "#FF0000"' '\n'
            'B.color = "#FF0000"' '\n'  # line break, so this shouldn't warn with --check-...=no
        )

    on_the_other_hand = ("this"
        "works fine with check=no")
```

### Configuration

```ini
--check-str-concat-over-line-jumps=no
```

### Command used

```shell
pylint a.py --check-str-concat-over-line-jumps=no
```

### Pylint output

```shell
a.py:7:0: W1404: Implicit string concatenation found in call (implicit-str-concat)
```

### Expected behavior

no `implict-str-concat` msg

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.12.0-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by jacobtylerwalls** [comment]
Ah, this report is invalid. There was a same-line implicit concat under my nose that I didn't see between `"#FF0000"' '\n'`.

**Comment by Pierre-Sassoulas** [comment]
>  There was a same-line implicit concat under my nose

Yeah, this check is really useful, these bugs are really hard to spot. We can reuse this example for #6444
